lib/mysql/mysql.go

```diff
@@ -0,0 +1,651 @@
+	IsSecure bool
+
+	// List of MySQL packets received/sent
+	PacketLog []*PacketLogEntry
+	// The zgrab TLS handshake logs
+	TLSHandshake *tls.ServerHandshake
+}
+
+// Constructor, filling in defaults where needed
+func NewConnection(config *Config) *Connection {
+	return &Connection{
+		Config:         NewConfig(config),
+		State:          STATE_NOT_CONNECTED,
+		PacketLog:      nil,
+		Connection:     nil,
+		SequenceNumber: 0}
+}
+
+// Top-level interface for all packets
+type PacketInfo interface {
+	GetType() PacketType
+	GetDescription() string
+}
+
+// Most packets are read from the server; for packets that need to be sent, they need an encoding function
+type WritablePacket interface {
+	PacketInfo
+	EncodeBody() []byte
+}
+
+// Entry in the PacketLog. Raw is the base64-encoded body, Parsed is the parsed packet.
+// Either may be nil if there was an error reading/decoding the packet.
+type PacketLogEntry struct {
+	Length         uint32     `json:"length"`
+	SequenceNumber uint8      `json:"sequence_number"`
+	Raw            string     `json:"raw"`
+	Parsed         PacketInfo `json:"parsed,omitempty"`
+}
+
+// Handshake_Packet defined at https://web.archive.org/web/20160316105725/https://dev.mysql.com/doc/internals/en/connection-phase-packets.html
+// @TODO @FIXME: This is protocol version 10; handle previous / future versions
+type Handshake_Packet struct {
+	// protocol_version: int<1>
+	ProtocolVersion byte `json:"protocol_version"`
+	// server_version: string<NUL>
+	ServerVersion string `json:"server_version"`
+	// connection_id: int<4>
+	ConnectionID uint32 `json:"connection_id"` // [4]
+	// auth_plugin_data_part_1: string<8>
+	AuthPluginData1 string `json:"auth_plugin_data_part_1"`
+	// fillter_1: byte<1>
+	Filler1 byte `json:"filler_1,omitempty"`
+	// capability_flag_1: int<2> -- Stored as lower 16 bits of capability_flags
+	// character_set: int<1> (optional?)
+	CharacterSet byte `json:"character_set"`
+
+	// Synthetic field: if true, none of the following fields are present.
+	ShortHandshake bool `json:"short_handshake"`
+
+	// status_flags: int<> (optional?)
+	StatusFlags uint16 `json:"status_flags"`
+	// capability_flag_2: int<2> -- Stored as upper 16 bits of capability_flags
+
+	// auth_plugin_data_len: int<1>
+	AuthPluginDataLen byte `json:"auth_plugin_data_len"`
+	// if (capabilities & CLIENT_SECURE_CONNECTION) {
+	// reserved:  string<10> all 0
+	Reserved []byte `json:"reserved,omitempty"`
+	// auth_plugin_data_part_2: string<MAX(13, auth_plugin_data_len - 8)>
+	AuthPluginData2 string `json:"auth_plugin_data_part_2,omitempty"`
+	// auth_plugin_name: string<NUL>, but old versions lacked null terminator, so returning string<EOF>
+	AuthPluginName string `json:"auth_plugin_name,omitempty"`
+	// }
+	// Synthetic field buily from capability_flags_1 || capability_flags_2 << 16
+	CapabilityFlags uint32 `json:"capability_flags"`
+}
+
+func (p *Handshake_Packet) GetDescription() string {
```

Comment:
I'm not sure I follow why this exists. What's the goal with `GetDescription()`?

Reply:
After looking at some other zgrab code, it probably shouldn’t. The JSON attributes in the strict definition achieve what I was going for here.

- [ ] Remove